Tests/SentryTests/Integrations/Performance/IO/SentryFileIOTrackingIntegrationObjCTests.m

```diff
@@ -0,0 +1,170 @@
+        [self assertData:[NSData dataWithContentsOfFile:self->filePath
+                                                options:NSDataReadingUncached
+                                                  error:nil]];
+    }];
+}
+
+- (void)test_dataWithContentsOfURL
+{
+    [self assertTransaction:^{ [self assertData:[NSData dataWithContentsOfURL:self->fileUrl]]; }];
+}
+
+- (void)test_dataWithContentsOfURLOptionsError
+{
+    [self assertTransaction:^{
+        [self assertData:[NSData dataWithContentsOfURL:self->fileUrl
+                                               options:NSDataReadingUncached
+                                                 error:nil]];
+    }];
+}
+
+- (void)test_initWithContentsOfURL
+{
+    [self assertTransaction:^{
+        [self assertData:[[NSData alloc] initWithContentsOfURL:self->fileUrl]];
+    }];
+}
+
+- (void)test_initWithContentsOfFile
+{
+    [self assertTransaction:^{
+        [self assertData:[[NSData alloc] initWithContentsOfFile:self->filePath]];
+    }];
+}
+
+- (void)test_writeToFileAtomically
+{
+    [self assertTransaction:^{ [self->someData writeToFile:self->filePath atomically:true]; }];
+    [self assertDataWritten];
+}
+
+- (void)test_writeToUrlAtomically
+{
+    [self assertTransaction:^{ [self->someData writeToURL:self->fileUrl atomically:true]; }];
+    [self assertDataWritten];
+}
+
+- (void)test_writeToFileOptionsError
+{
+    [self assertTransaction:^{
+        [self->someData writeToFile:self->filePath options:NSDataWritingAtomic error:nil];
+    }];
+    [self assertDataWritten];
+}
+
+- (void)test_writeToUrlOptionsError
+{
+    [self assertTransaction:^{
+        [self->someData writeToURL:self->fileUrl options:NSDataWritingAtomic error:nil];
+    }];
+    [self assertDataWritten];
+}
+
+- (void)test_NSFileManagerContentAtPath
+{
+    [self assertTransaction:^{
+        [self assertData:[NSFileManager.defaultManager contentsAtPath:self->filePath]];
+    }];
+}
+
+- (void)test_NSFileManagerCreateFile
+{
+    [self assertTransaction:^{
+        [NSFileManager.defaultManager createFileAtPath:self->filePath
+                                              contents:self->someData
+                                            attributes:nil];
+    }];
+    [self assertDataWritten];
+}
+
+- (void)assertData:(NSData *)data
+{
+    NSString *content = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+
+    XCTAssertTrue([content isEqualToString:@"SOME DATA"]);
+    XCTAssertEqual(data.length, 9);
+}
+
+- (void)assertDataWritten
+{
+    [self assertData:[NSData dataWithContentsOfFile:filePath]];
+}
+
+- (void)assertTransaction:(void (^)(void))block
+{
+    SentryTracer *parentTransaction = [SentrySDK startTransactionWithName:@"Transaction"
+                                                                operation:@"Test"
+                                                              bindToScope:YES];
+
+    block();
+
+    XCTAssertEqual(parentTransaction.children.count, 1);
```

Comment:
`l`: I think it would be easy to also assert the description and the operation of the file I/O span in here. This would improve the test a bit.

Reply:
Sure. I'll look into it.